### ISSUE TITLE
Status is dropped when translating spans to Jaeger TChannel spans

### ISSUE THREAD

**Issue by asutoshpalai** [census-instrumentation/opencensus-service#586]
The status field and its message is not added to the spans as tags while translating the spans from the OpenCensus span format to TChannel span format at [translator/trace/jaeger/protospan_to_jaegerthrift.go#L148](https://github.com/census-instrumentation/opencensus-service/blob/dee9655f45a480266d5ab80ef206b5faf97c2f04/translator/trace/jaeger/protospan_to_jaegerthrift.go#L148).

This is done correctly in `protospan_to_jaegerproto.go` and Jaeger exporter for OpenCensus.

**Comment by pjanotti** [comment]
Thanks for reporting @asutoshpalai. /cc @owais

**Comment by owais** [comment]
We should have a fix for this in a few days @asutoshpalai

**Comment by asutoshpalai** [comment]
@owais Have you started working on this? If not, can I please pick this up?

**Comment by owais** [comment]
@asutoshpalai Yes, I'm almost done with it. Will submit a patch in a couple of days.

**Comment by asutoshpalai** [comment]
I was going through the code of jaegerproto, it looks like they don't set `error` tag when the `status` is  not `OK`. This is not consistent with the behavior of Jaeger exporter for OpenCensus which does that, [census-ecosystem/opencensus-go-exporter-jaeger/blob/master/jaeger.go#L217](https://github.com/census-ecosystem/opencensus-go-exporter-jaeger/blob/540daef1da72376e9b3e1d52e4f90403d72f5c00/jaeger.go#L215).

Ref: census-instrumentation/opencensus-go#1041

I would really appreciate it if @owais can include that in his fix for thrift protocol too.

**Comment by pjanotti** [comment]
One important observation here: the behavior on opencensus-go is not the recommended one for OpenTracing (which Jaeger follows). The error tag should be explictly set by the caller code. I had mentioned this on the opencensus-go repo and this was recently changed on java client library. Relevant links:

**Comment by owais** [comment]
We have a PR for OT here: https://github.com/open-telemetry/opentelemetry-service/pull/70

Once it is accepted there, I'll back-port the fix to OC.

**Comment by owais** [comment]
We've merged it to OT. Will try to backport here later this week.
